### ISSUE TITLE
Conditions and regex for tags

### ISSUE THREAD

**Issue by tamtakoe** [GoogleContainerTools/jib#2259]
General configuration
```
<version>1.2.3-SNAPSHOT</version>
...
<plugin>
  <groupId>com.google.cloud.tools</groupId>
  <artifactId>jib-maven-plugin</artifactId>
  <version>${jib-maven-plugin.version}</version>
  <configuration>
    <to>
      <tags>
        <tag>latest</tag>
        <tag>${GIT_LOCAL_BRANCH}</tag>
        </tags>
      </to>
    </configuration>
</plugin>
```
This is OK, but not suitable for many cases. The more productive way is to have tag with`1.2.3` for `master` and e.g. `1.2.3-my-branch-SNAPSHOT` for `my/branch`. Or something like this. We also need replace `/` to `-` because slash is incorrect for Docker naming etc.

So. It will be good to have more flexible tag settings. E.g.
```
<tags>
  <tag if="${GIT_LOCAL_BRANCH}==master">${project.version}</tag>
  <tag if="${GIT_LOCAL_BRANCH}!=master" regex="(.+)(-SNAPSHOT-)(.+)" replacement="$1-$3-SNAPSHOT" exclude="/">${project.version}-${GIT_LOCAL_BRANCH}</tag>
</tags>
```

**Comment by loosebazooka** [comment]
@tamtakoe this is probably too specific to be in the scope of jib's responsibilities. Could you maybe calculate this before and load the value into jibs config -- using something like the Maven properties plugin?

**Comment by rupert-madden-abbott** [comment]
@loosebazooka I agree that jib probably shouldn't offer specific extension points for VCS.

However, isn't there a problem that is going to affect all users of the plugin here? If you want to create specific tags for each -SNAPSHOT deploy then there is no way to supply that, only for -SNAPSHOT versions.

If I have retrieved a unique build number, commit id or timestamp from elsewhere, then it would be helpful to be able to attach it only to SNAPSHOT versions.

It is entirely possible to do this outside of jib also, but appending some unique information to SNAPSHOT versions seems like a much more common problem that many users of this plugin will have, as opposed to exactly which information should be appended in this way.
